### DISCUSSION TITLE
Suggestion to show iCloud account in settings

### DISCUSSION THREAD

**Discussion by kmatzka** [tact/public#117] [Ideas]
Hi Jaanus,

as discussed on Mastodon I make a note of my suggestion here:

Frist let me state: I really like the smoothness of onboarding to Tact with my iCloud account!

As I have two iCloud accounts, my user account and my dev account, I have been wondering which iCloud account is used for the onboarding. And for sure it can only be the one set as active account in settings in macOS or iOS.

I wanted to double check in settings whether it really is that account. But it is not shown there.
My suggestion: would be great to have a way of confirming with which account you are logged into Tact.

Follow up question comes to my mind: why not offer to choose the iCloud account that Tact uses (independently of the currently active one for the system)? Would this be technically possible? I guess so.

Best, Klaus.

**Comment by jaanus** [comment]
Thank you for the kind words! Indeed, one benefit of using iCloud account is that the onboarding is smooth for the user. There is no account or password to create, we don’t need to bother with things like “lost password” etc.

Showing the currently used iCloud account in Tact settings makes a lot of sense, thank you. I will put this in the development queue.

As far as choosing the account goes, Apple does not give apps like Tact any control over that. We just get iCloud access from the operating system, and can talk to the current account’s iCloud database, but that’s it. So, it’s not possible to build iCloud account picking in Tact.

I note that macOS and iOS iCloud UX is designed mostly for the case of one account. It’s possible to switch accounts by logging out of one and into another, but it’s a clumsy, awkward process.

When you do switch iCloud accounts, Tact picks that up correctly, removes local data for the previous account and works with the new account.

**Comment by jaanus** [comment]
I looked at this a bit. I thought there is a way to get the current iCloud account info like e-mail without doing anything dirty, but … no. CloudKit does not provide current iCloud user's e-mail address or any other account info to us with any API that I could think of. So, cannot implement this in Tact for the time being.

**Reply by kmatzka** [reply]
I came to the same conclusion. Seems it is not possible.

What I have been able to do is get the personal details including email of the "Me" card in Contacts by asking for permissions for userDiscoverability and access to contacts, getting contactIdentifier(s) of the current iCloud user and then searched Contacts with such contactIdentifier(s). But that is a totally different thing and not what the original idea has been.

**Reply by jaanus** [reply]
Yep. I’d like to stay away from userDiscoverability if I can. So far quite proud that Tact doesn’t use it.
